From 66dd2280b3ecbb4a09f4f262d52a3a99a97afd45 Mon Sep 17 00:00:00 2001
From: Petko Bordjukov <bordjukov@gmail.com>
Date: Mon, 25 Jul 2016 00:00:19 +0300
Subject: [PATCH 1/4] Build a standalone static library

---
 gypfiles/standalone.gypi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gypfiles/standalone.gypi b/gypfiles/standalone.gypi
index d438a5aeab..f28c9bfa9f 100644
--- a/gypfiles/standalone.gypi
+++ b/gypfiles/standalone.gypi
@@ -538,6 +538,9 @@
       }],  # fastbuild!=0
     ],
     'target_conditions': [
+      ['_type=="static_library"', {
+        'standalone_static_library': 1,
+      }],
       ['v8_code == 0', {
         'defines!': [
           'DEBUG',
-- 
2.12.1

