From 68eadedb9728ee6ce634d646835ef0aab2d23be6 Mon Sep 17 00:00:00 2001
From: Petko Bordjukov <bordjukov@gmail.com>
Date: Mon, 25 Jul 2016 00:05:47 +0300
Subject: [PATCH 3/5] Use the -fPIC flag for the static library

---
 gypfiles/standalone.gypi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gypfiles/standalone.gypi b/gypfiles/standalone.gypi
index d0d00d3f67..f01bd48596 100644
--- a/gypfiles/standalone.gypi
+++ b/gypfiles/standalone.gypi
@@ -797,7 +797,7 @@
           [ 'visibility=="hidden" and v8_enable_backtrace==0', {
             'cflags': [ '-fvisibility=hidden' ],
           }],
-          [ 'component=="shared_library"', {
+          [ 'component=="shared_library" or component=="static_library" and (v8_target_arch=="x64" or v8_target_arch=="arm64" or v8_target_arch=="arm")', {
             'cflags': [ '-fPIC', ],
           }],
           [ 'clang==0 and coverage==1', {
-- 
2.15.1

