From 22cd3f03af2d94da1f7506a8daa7708e8c93b928 Mon Sep 17 00:00:00 2001
From: Petko Bordjukov <bordjukov@gmail.com>
Date: Mon, 25 Jul 2016 00:05:47 +0300
Subject: [PATCH 3/3] Use the -fPIC flag for the static library

---
 gypfiles/standalone.gypi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gypfiles/standalone.gypi b/gypfiles/standalone.gypi
index 1a69c6f..b19304f 100644
--- a/gypfiles/standalone.gypi
+++ b/gypfiles/standalone.gypi
@@ -744,7 +744,7 @@
           [ 'visibility=="hidden" and v8_enable_backtrace==0', {
             'cflags': [ '-fvisibility=hidden' ],
           }],
-          [ 'component=="shared_library"', {
+          [ 'component=="shared_library" or component=="static_library" and (v8_target_arch=="x64" or v8_target_arch=="arm64" or v8_target_arch=="arm")', {
             'cflags': [ '-fPIC', ],
           }],
           [ 'clang==0 and coverage==1', {
-- 
2.9.0

