From 4792a3b2cca0ac92fef185c4f7394e3580d8ac23 Mon Sep 17 00:00:00 2001
From: Petko Bordjukov <bordjukov@gmail.com>
Date: Mon, 25 Jul 2016 00:05:47 +0300
Subject: [PATCH 3/4] Use the -fPIC flag for the static library

---
 gypfiles/standalone.gypi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gypfiles/standalone.gypi b/gypfiles/standalone.gypi
index c81ae2caa0..b3e4d67a6e 100644
--- a/gypfiles/standalone.gypi
+++ b/gypfiles/standalone.gypi
@@ -803,7 +803,7 @@
           [ 'visibility=="hidden" and v8_enable_backtrace==0', {
             'cflags': [ '-fvisibility=hidden' ],
           }],
-          [ 'component=="shared_library"', {
+          [ 'component=="shared_library" or component=="static_library" and (v8_target_arch=="x64" or v8_target_arch=="arm64" or v8_target_arch=="arm")', {
             'cflags': [ '-fPIC', ],
           }],
           [ 'clang==0 and coverage==1', {
-- 
2.11.1

