From c04e8c3d31921526dfcb726a79b92c55d22713bf Mon Sep 17 00:00:00 2001
From: Petko Bordjukov <bordjukov@gmail.com>
Date: Wed, 27 Jul 2016 06:20:13 +0300
Subject: [PATCH 4/5] Reinterpret thread hash for FreeBSD, too

---
 src/libsampler/v8-sampler.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libsampler/v8-sampler.cc b/src/libsampler/v8-sampler.cc
index edf6df130e..9b2a71893b 100644
--- a/src/libsampler/v8-sampler.cc
+++ b/src/libsampler/v8-sampler.cc
@@ -194,7 +194,7 @@ void* ThreadKey(pthread_t thread_id) {
 
 // Returns hash value for hash map.
 uint32_t ThreadHash(pthread_t thread_id) {
-#if V8_OS_MACOSX
+#if (V8_OS_MACOSX || V8_OS_FREEBSD)
   return static_cast<uint32_t>(reinterpret_cast<intptr_t>(thread_id));
 #else
   return static_cast<uint32_t>(thread_id);
-- 
2.11.1

