From cf7ec827796ccaef75e77bd83771e7f31b645d2f Mon Sep 17 00:00:00 2001
From: Petko Bordjukov <bordjukov@gmail.com>
Date: Wed, 27 Jul 2016 06:20:13 +0300
Subject: [PATCH 4/4] Reinterpret thread hash for FreeBSD, too

---
 src/libsampler/v8-sampler.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libsampler/v8-sampler.cc b/src/libsampler/v8-sampler.cc
index edf6df1..9b2a718 100644
--- a/src/libsampler/v8-sampler.cc
+++ b/src/libsampler/v8-sampler.cc
@@ -194,7 +194,7 @@ void* ThreadKey(pthread_t thread_id) {
 
 // Returns hash value for hash map.
 uint32_t ThreadHash(pthread_t thread_id) {
-#if V8_OS_MACOSX
+#if (V8_OS_MACOSX || V8_OS_FREEBSD)
   return static_cast<uint32_t>(reinterpret_cast<intptr_t>(thread_id));
 #else
   return static_cast<uint32_t>(thread_id);
-- 
2.9.3

