diff --git a/Makefile b/Makefile
index d04d9ef..bd60399 100644
--- a/Makefile
+++ b/Makefile
@@ -240,7 +240,7 @@ NACL_ARCHES = nacl_ia32 nacl_x64
 # List of files that trigger Makefile regeneration:
 GYPFILES = build/all.gyp build/features.gypi build/standalone.gypi \
            build/toolchain.gypi samples/samples.gyp src/d8.gyp \
-           test/cctest/cctest.gyp test/unittests/unittests.gyp tools/gyp/v8.gyp
+           tools/gyp/v8.gyp
 
 # If vtunejit=on, the v8vtune.gyp will be appended.
 ifeq ($(vtunejit), on)
diff --git a/build/all.gyp b/build/all.gyp
index 4aeb507..96820a0 100644
--- a/build/all.gyp
+++ b/build/all.gyp
@@ -8,10 +8,7 @@
       'target_name': 'All',
       'type': 'none',
       'dependencies': [
-        '../samples/samples.gyp:*',
         '../src/d8.gyp:d8',
-        '../test/cctest/cctest.gyp:*',
-        '../test/unittests/unittests.gyp:*',
       ],
       'conditions': [
         ['component!="shared_library"', {
