diff --git a/Makefile b/Makefile
index 0d825c0..c508f33 100644
--- a/Makefile
+++ b/Makefile
@@ -95,6 +95,11 @@ endif
 ifeq ($(strictaliasing), off)
   GYPFLAGS += -Dv8_no_strict_aliasing=1
 endif
+# hardfp=on
+ifeq ($(hardfp), on)
+  GYPFLAGS += -Dv8_use_arm_eabi_hardfloat=true
+endif
+
 
 # ----------------- available targets: --------------------
 # - "dependencies": pulls in external dependencies (currently: GYP)
diff --git a/build/common.gypi b/build/common.gypi
index 7f084b8..d89f1f4 100644
--- a/build/common.gypi
+++ b/build/common.gypi
@@ -133,7 +133,6 @@
           [ 'v8_use_arm_eabi_hardfloat=="true"', {
             'defines': [
               'USE_EABI_HARDFLOAT=1',
-              'CAN_USE_VFP_INSTRUCTIONS',
             ],
             'target_conditions': [
               ['_toolset=="target"', {
diff --git a/build/standalone.gypi b/build/standalone.gypi
index ebdf557..2961bac 100644
--- a/build/standalone.gypi
+++ b/build/standalone.gypi
@@ -74,9 +74,9 @@
       }],
     ],
     # Default ARM variable settings.
-    'armv7%': 1,
+    'armv7%': 0,
     'arm_neon%': 0,
-    'arm_fpu%': 'vfpv3',
+    'arm_fpu%': 'vfp',
   },
   'target_defaults': {
     'default_configuration': 'Debug',
