diff --git a/build/standalone.gypi b/build/standalone.gypi
index ebdf557..c7a59bc 100644
--- a/build/standalone.gypi
+++ b/build/standalone.gypi
@@ -98,6 +98,9 @@
           [ 'OS=="linux"', {
             'cflags': [ '-ansi' ],
           }],
+          [ '(OS=="linux" or OS=="freebsd") and v8_target_arch=="x64" and component=="static_library"', {
+            'cflags': [ '-fPIC' ],
+          }],
           [ 'visibility=="hidden"', {
             'cflags': [ '-fvisibility=hidden' ],
           }],
