diff --git a/Makefile b/Makefile
index f49be61..0626d02 100644
--- a/Makefile
+++ b/Makefile
@@ -463,13 +463,5 @@ gtags.clean:
 # Dependencies.
 # Remember to keep these in sync with the DEPS file.
 dependencies:
-	svn checkout --force http://gyp.googlecode.com/svn/trunk build/gyp \
-	    --revision 1831
-	svn checkout --force \
-	    https://src.chromium.org/chrome/trunk/deps/third_party/icu46 \
-	    third_party/icu --revision 258359
-	( test -d buildtools || \
-	  git clone https://chromium.googlesource.com/chromium/buildtools.git; \
-	  cd buildtools; \
-	  git fetch origin; \
-	  git checkout 5d89977ce55240995d1596fe420b818468f5ec37 )
+	( test -h build/gyp || ln -s ../../gyp build/gyp )
+	( test -h buildtools || ln -s ../buildtools )
