diff --git a/src/platform-freebsd.cc b/src/platform-freebsd.cc
index a1a0739..f21e254 100644
--- a/src/platform-freebsd.cc
+++ b/src/platform-freebsd.cc
@@ -121,10 +121,10 @@ static unsigned StringToLong(char* buffer) {
 
 
 std::vector<OS::SharedLibraryAddress> OS::GetSharedLibraryAddresses() {
-  std::vector<SharedLibraryAddress> result;
+  std::vector<SharedLibraryAddress> result2;
   static const int MAP_LENGTH = 1024;
   int fd = open("/proc/self/maps", O_RDONLY);
-  if (fd < 0) return result;
+  if (fd < 0) return result2;
   while (true) {
     char addr_buffer[11];
     addr_buffer[0] = '0';
@@ -155,10 +155,10 @@ std::vector<OS::SharedLibraryAddress> OS::GetSharedLibraryAddresses() {
     // There may be no filename in this line.  Skip to next.
     if (start_of_path == NULL) continue;
     buffer[bytes_read] = 0;
-    result.push_back(SharedLibraryAddress(start_of_path, start, end));
+    result2.push_back(SharedLibraryAddress(start_of_path, start, end));
   }
   close(fd);
-  return result;
+  return result2;
 }
 
 
